src/core/hle/libraries/libkernel/thread_management.cpp

```diff
@@ -0,0 +1,298 @@
+}
+
+int PS4_SYSV_ABI scePthreadMutexattrSettype(ScePthreadMutexattr* attr, int type) {
+    int ptype = PTHREAD_MUTEX_DEFAULT;
+    switch (type) {
+        case 1: ptype = PTHREAD_MUTEX_ERRORCHECK; break;
+        case 2: ptype = PTHREAD_MUTEX_RECURSIVE; break;
+        case 3:
+        case 4: ptype = PTHREAD_MUTEX_NORMAL; break;
+        default: LOG_TRACE_IF(log_pthread_file, "scePthreadMutexattrSettype invalid type: {}\n", type); std::exit(0);
+    }
+
+    int result = pthread_mutexattr_settype(&(*attr)->pth_mutex_attr, ptype);
+
+    return result == 0 ? SCE_OK : SCE_KERNEL_ERROR_EINVAL;
+}
+
+int PS4_SYSV_ABI scePthreadMutexattrSetprotocol(ScePthreadMutexattr* attr, int protocol) {
+    int pprotocol = PTHREAD_PRIO_NONE;
+    switch (protocol) {
+        case 0: pprotocol = PTHREAD_PRIO_NONE; break;
+        case 1: pprotocol = PTHREAD_PRIO_INHERIT; break;
+        case 2: pprotocol = PTHREAD_PRIO_PROTECT; break;
+        default: LOG_TRACE_IF(log_pthread_file, "scePthreadMutexattrSetprotocol invalid protocol: {}\n", protocol); std::exit(0);
+    }
+
+    int result = 0;  // pthread_mutexattr_setprotocol(&(*attr)->p, pprotocol); //it appears that pprotocol has issues in winpthreads
+    (*attr)->pprotocol = pprotocol;
+
+    return result == 0 ? SCE_OK : SCE_KERNEL_ERROR_EINVAL;
+}
+int PS4_SYSV_ABI scePthreadMutexLock(ScePthreadMutex* mutex) {
+    PRINT_FUNCTION_NAME();
+    mutex = static_cast<ScePthreadMutex*>(createMutex(mutex));
+
+    if (mutex == nullptr) {
+        return SCE_KERNEL_ERROR_EINVAL;
+    }
+
+    int result = pthread_mutex_lock(&(*mutex)->pth_mutex);
+    LOG_INFO_IF(log_pthread_file, "scePthreadMutexLock name={} result={}\n", (*mutex)->name, result);
+    switch (result) {
+        case 0: return SCE_OK;
+        case EAGAIN: return SCE_KERNEL_ERROR_EAGAIN;
+        case EINVAL: return SCE_KERNEL_ERROR_EINVAL;
+        case EDEADLK: return SCE_KERNEL_ERROR_EDEADLK;
+        default: return SCE_KERNEL_ERROR_EINVAL;
+    }
+}
+int PS4_SYSV_ABI scePthreadMutexUnlock(ScePthreadMutex* mutex) {
+    PRINT_FUNCTION_NAME();
+    mutex = static_cast<ScePthreadMutex*>(createMutex(mutex));
+    if (mutex == nullptr) {
+        return SCE_KERNEL_ERROR_EINVAL;
+    }
+
+    int result = pthread_mutex_unlock(&(*mutex)->pth_mutex);
+    LOG_INFO_IF(log_pthread_file, "scePthreadMutexUnlock name={} result={}\n", (*mutex)->name, result);
+    switch (result) {
+        case 0: return SCE_OK;
+
+        case EINVAL: return SCE_KERNEL_ERROR_EINVAL;
+        case EPERM: return SCE_KERNEL_ERROR_EPERM;
+        default: return SCE_KERNEL_ERROR_EINVAL;
+    }
+}
+
+int PS4_SYSV_ABI posix_pthread_mutex_init(ScePthreadMutex* mutex, const ScePthreadMutexattr* attr) {
+    LOG_INFO_IF(log_pthread_file, "posix pthread_mutex_init redirect to scePthreadMutexInit\n");
+    int result = scePthreadMutexInit(mutex, attr, nullptr);
+    if (result < 0) {
+        int rt = result > 0x80020000 && result <= 0x80020065 ? result + -0x80020000 : 1062;
```

Comment:
Εεε μας πας λιγο πηδιωντας με τα magic values

Reply:
done it better